### PR TITLE
Add view to confirm that student finished oral interview

### LINKED ISSUE #320: API view: oral interview successful, can wait for group
Some students (e.g. those that can't read or haven't learned English long enough) require a one-to-one interview. The student's language level is determined in this interview, then they can be put in the waiting queue.

A coordinator send from dashboard a request stating that the given `Student` has a given language level and can be put into the queue. The data in a `POST` request must include ID of `LanguageAndLevel` object.

## Check `Student`'s `project_status`

If the given `Student`'s `project_status` is not `NEEDS_INTERVIEW_TO_DETERMINE_LEVEL`, this is a wrong request.

## Change objects

Create a new many-to-many connection between `Student` and `LanguageAndLevel` by adding the `LanguageAndLevel` object to `.teaching_languages_and_levels`.

> [!NOTE] 
> Most likely all students will only be allowed to study in one group, but the database was designed to be more flexible and to technically allow several groups.

## Set status

Set `project_status` to `NO_GROUP_YET`

Remember to update `status_since` to current time in UTC in objects (where applicable).

## Create log events

Add log event of type `AWAITING_OFFER`
(**TODO:** check what event type is created when student is created through bot)

### PR DESCRIPTION
**Pull request by Auch-Auch**
Closes #320

### CODE REVIEW

File: api/views/student.py [hunk 1]
```diff
@@ -293,6 +294,36 @@ def active_students_with_no_groups(self, request: Request) -> Response:  # noqa:
...
 
+    @extend_schema(
+        request=DashboardFinishedOralInterviewSerializer,
+        responses={
+            status.HTTP_204_NO_CONTENT: OpenApiResponse(description="Action is taken"),
+            status.HTTP_409_CONFLICT: OpenApiResponse(
+                response=BaseAPIExceptionSerializer,
+                description="Only with status NEEDS_INTERVIEW_TO_DETERMINE_LEVEL can be processed",
+            ),
+            status.HTTP_422_UNPROCESSABLE_ENTITY: OpenApiResponse(
```
Comment (Auch-Auch) [inline api/views/student.py:305]: I suggest using this status everywhere where we have not found some entity, so use it instead of 409, who do you think?

Comment (lemontree210) [reply api/views/student.py:305]: yes, let's go for 422. I searched through old messages in our TG chat and see that @Roxe322 was in favor of 422 over 409 in similar cases. The main argument for 422 is that it means that the client [should not repeat the request without modificatio](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422), which is exactly what we want to tell the client. 409 doesn't necessary entail that.

**Review by lemontree210 — CHANGES_REQUESTED** [review]
I would ask you to replace "finished" with e.g. "completed" (in urls and class names). Finished can be confused with "finished studies".

I could also suggest "passed" but you can't actually fail this interview.

**Review by lemontree210 — APPROVED** [review]
LGTM. Thanks for being so thorough :)

**Pull request merged**